hive-metastore/src/test/java/org/apache/iceberg/hive/TestHiveClientPool.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.hive;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import org.apache.hadoop.hive.conf.HiveConf;
+import org.junit.Assert;
+import org.junit.Test;
+
+public class TestHiveClientPool {
+
+  private static final String HIVE_SITE_CONTENT = "<?xml version=\"1.0\"?>\n" +
+          "<?xml-stylesheet type=\"text/xsl\" href=\"configuration.xsl\"?>\n" +
+          "<configuration>\n" +
+          "  <property>\n" +
+          "    <name>hive.metastore.sasl.enabled</name>\n" +
+          "    <value>true</value>\n" +
+          "  </property>\n" +
+          "</configuration>\n";
+
+  @Test
+  public void testConf() {
+    HiveConf conf = createHiveConf();
+    conf.set(HiveConf.ConfVars.METASTOREURIS.varname, "thrift://locahost:12345");
+    conf.set(HiveConf.ConfVars.METASTOREWAREHOUSE.varname, "file:/mywarehouse/");
+    conf.setInt("iceberg.hive.client-pool-size", 10);
+
+    HiveClientPool clientPool = new HiveClientPool(conf);
+    HiveConf clientConf = clientPool.hiveConf();
+
+    // hive will pick up system property whenever a new HiveConf is created
+    // see org.apache.iceberg.hive.TestHiveMetastore.start(int)
+    if (System.getProperty(HiveConf.ConfVars.METASTOREURIS.varname) != null) {
+      Assert.assertEquals(System.getProperty(HiveConf.ConfVars.METASTOREURIS.varname),
+              clientConf.get(HiveConf.ConfVars.METASTOREURIS.varname));
+    } else {
+      Assert.assertEquals(conf.get(HiveConf.ConfVars.METASTOREURIS.varname),
+              clientConf.get(HiveConf.ConfVars.METASTOREURIS.varname));
+    }
```

Comment:
Do we want to test this case specifically, or it is just happens that this System property is set in some environments and want to avoid this situation. If this is the later, we might just use `System.clearProperty(HiveConf.ConfVars.METASTOREURIS.varname)` to make sure that this not interferes with our tests.

Reply:
Thanks peter, this test branch lead misunderstanding, it's totaly has nothing to do with this PR, i will delete it.